### PR TITLE
chore(chat): refactor chat component

### PR DESCRIPTION
**Pull request by sergesha**
**Description:**

Refactor <Chat/> component by splitting the code into sub-components and custom hooks to enhance readability and maintainability

Issues:

- No #<TICKET_ID>

**UI changes**

- No UI changes

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
Unknown release type "refactor" found in pull request title "refactor(chat): refactor chat component". 

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status:

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1950.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/512997)

File: package.json [hunk 1]
```diff
@@ -168,5 +170,6 @@
     "fs": false,
     "net": false,
     "tls": false
-  }
+  },
+  "packageManager": "yarn@1.22.19+sha1.4ba7fc5c6e704fce2066ecbfb0b0d8976fe62447"
```
Comment (IlyaBondar) [inline package.json]: I think we don't need this

File: apps/chat/src/components/Chat/AssistantSubModelSelector.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { ModelsSelectors } from '@/src/store/models/models.reducers';
 
 import { Combobox } from '../Common/Combobox';
 import { DisableOverlay } from '../Common/DisableOverlay';
-import { ModelSelectRow } from './ConversationSettings';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/AssistantSubModelSelector.tsx:16]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/ChatInput/ChatControls.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ import { SettingsSelectors } from '@/src/store/settings/settings.reducers';
 import { SendMessageButton } from '@/src/components/Chat/ChatInput/SendMessageButton';
 import Tooltip from '@/src/components/Common/Tooltip';
 
-import RefreshCW from '../../../../public/images/icons/refresh-cw.svg';
+import RefreshCW from '@/public/images/icons/refresh-cw.svg';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatControls.tsx:20]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

File: apps/chat/src/components/Chat/ChatInput/ChatInput.tsx [hunk 4]
```diff
@@ -5,7 +5,8 @@ import { Message } from '@/src/types/chat';
 import { ConversationsSelectors } from '@/src/store/conversations/conversations.reducers';
 import { useAppSelector } from '@/src/store/hooks';
 
-import { ChatInputFooter } from './ChatInputFooter';
+import { ChatInputFooter } from '@/src/components/Chat/common/ChatInputFooter';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatInput.tsx:8]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/ChatInput/ChatInputMessage.tsx [hunk 5]
```diff
@@ -35,12 +35,12 @@ import { UISelectors } from '@/src/store/ui/ui.reducers';
 
 import { errorsMessages } from '@/src/constants/errors';
 
-import { ChatControls } from '@/src/components/Chat/ChatInput/ChatControls';
+import { AdjustedTextarea } from '@/src/components/Common/AdjustedTextarea';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatInputMessage.tsx:38]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

File: apps/chat/src/components/Chat/ChatView/ChatCompareSelect.tsx [hunk 6]
```diff
@@ -15,10 +15,10 @@ import { ConversationsSelectors } from '@/src/store/conversations/conversations.
 import { useAppSelector } from '@/src/store/hooks';
 import { ModelsSelectors } from '@/src/store/models/models.reducers';
 
-import { ModelIcon } from '../Chatbar/ModelIcon';
-import { Combobox } from '../Common/Combobox';
-import Loader from '../Common/Loader';
-import ShareIcon from '../Common/ShareIcon';
+import { ModelIcon } from '@/src/components/Chatbar/ModelIcon';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatView/ChatCompareSelect.tsx:18]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/ChatView/ChatMessages/ChatMessage.tsx [hunk 7]
```diff
@@ -11,11 +11,12 @@ import { Translation } from '@/src/types/translation';
 import { useAppSelector } from '@/src/store/hooks';
 import { SettingsSelectors } from '@/src/store/settings/settings.reducers';
 
-import { ChatMessageContent } from '@/src/components/Chat/ChatMessage/ChatMessageContent';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatView/ChatMessages/ChatMessage.tsx:14]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/ChatView/ChatMessages/ChatMessageContent.tsx [hunk 8]
```diff
@@ -40,18 +40,15 @@ import { UISelectors } from '@/src/store/ui/ui.reducers';
...
 
 import { ChatInputAttachments } from '@/src/components/Chat/ChatInput/ChatInputAttachments';
-import {
-  MessageAssistantButtons,
-  MessageUserButtons,
-} from '@/src/components/Chat/ChatMessage/MessageButtons';
 import { MessageAttachments } from '@/src/components/Chat/MessageAttachments';
 import { MessageStages } from '@/src/components/Chat/MessageStages';
 import { ModelIcon } from '@/src/components/Chatbar/ModelIcon';
+import { AdjustedTextarea } from '@/src/components/Common/AdjustedTextarea';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatView/ChatMessages/ChatMessageContent.tsx:46]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

File: apps/chat/src/components/Chat/ChatView/ChatSettingsEmpty.tsx [hunk 9]
```diff
@@ -1,45 +1,46 @@
...
 import { Prompt } from '@/src/types/prompt';
 
 import { useAppSelector } from '@/src/store/hooks';
-import { ModelsSelectors } from '@/src/store/models/models.reducers';
 import { SettingsSelectors } from '@/src/store/settings/settings.reducers';
 
 import { REPLAY_AS_IS_MODEL } from '@/src/constants/chat';
 
-import { Spinner } from '../Common/Spinner';
+import { Spinner } from '../../Common/Spinner';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatView/ChatSettingsEmpty.tsx:12]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/ChatView/ConversationSettings.tsx [hunk 10]
```diff
@@ -24,16 +24,16 @@ import {
 } from '@/src/constants/chat';
 import { DEFAULT_ASSISTANT_SUBMODEL_ID } from '@/src/constants/default-ui-settings';
 
+import { TemperatureSlider } from './components/Temperature';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatView/ConversationSettings.tsx:27]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

File: apps/chat/src/components/Chat/MessageAttachment.tsx [hunk 11]
```diff
@@ -26,12 +26,12 @@ import { FOLDER_ATTACHMENT_CONTENT_TYPE } from '@/src/constants/folders';
 import { Spinner } from '@/src/components/Common/Spinner';
 import { PlotlyComponent } from '@/src/components/Plotly/Plotly';
 
-import Link from '../../../public/images/icons/arrow-up-right-from-square.svg';
-import ChevronDown from '../../../public/images/icons/chevron-down.svg';
 import Tooltip from '../Common/Tooltip';
 import ChatMDComponent from '../Markdown/ChatMDComponent';
 import { VisualizerRenderer } from '../VisualalizerRenderer/VisualizerRenderer';
 
+import Link from '@/public/images/icons/arrow-up-right-from-square.svg';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/MessageAttachment.tsx:33]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

File: apps/chat/src/components/Chat/MessageStage.tsx [hunk 12]
```diff
@@ -11,12 +11,13 @@ import { useAppSelector } from '@/src/store/hooks';
 
 import { ModelIcon } from '@/src/components/Chatbar/ModelIcon';
 
-import ChevronDown from '../../../public/images/icons/chevron-down.svg';
-import CircleCheck from '../../../public/images/icons/circle-check.svg';
 import { Spinner } from '../Common/Spinner';
 import ChatMDComponent from '../Markdown/ChatMDComponent';
 import { MessageAttachments } from './MessageAttachments';
 
+import ChevronDown from '@/public/images/icons/chevron-down.svg';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/MessageStage.tsx:18]: this component still use state directly (`useAppSelector`)

File: apps/chat/src/components/Chat/Playback/PlaybackControls.tsx [hunk 13]
```diff
@@ -21,9 +21,9 @@ import {
 import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
 import { UISelectors } from '@/src/store/ui/ui.reducers';
 
+import { ChatInputFooter } from '@/src/components/Chat/common/ChatInputFooter';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Playback/PlaybackControls.tsx:24]: this component still use state directly (`useAppSelector`, `useAppDispatch`)

**Comment by IlyaBondar** [comment]
Main problem of this approach is **prop drilling** from top parent component to target child component. We collect a lot of data and handlers on top level into big object and than drill them down to target. On any changes data-object will be new - **rerender**.
If we need new property on lowest level we need to drill it down from all levels between top and lowest level. I.e. will be problem with **supporting and maintaining this code**.

At this moment the goal to make pure Chat-component wasn't achieved because a lot of child components are using state directly (`useAppSelector`, `useAppDispatch`)

**Pull request closed without merging**